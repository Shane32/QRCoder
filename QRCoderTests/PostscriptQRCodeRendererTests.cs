namespace QRCoderTests;

public class PostscriptQRCodeRendererTests
{
    [Fact]
    public void can_render_postscript_qrcode_simple()
    {
        //Create QR code
        var gen = new QRCodeGenerator();
        var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.L);
        var ps = new PostscriptQRCode(data).GetGraphic(5);
        RemoveCreationDate(ps).ShouldMatchApproved(x => x.NoDiff());
    }

    [Fact]
    public void can_render_postscript_qrcode_eps()
    {
        //Create QR code
        var gen = new QRCodeGenerator();
        var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.L);
        var ps = new PostscriptQRCode(data).GetGraphic(5, true);
        RemoveCreationDate(ps).ShouldMatchApproved(x => x.NoDiff());
    }

    [Fact]
    public void can_render_postscript_qrcode_size()
    {
        //Create QR code
        var gen = new QRCodeGenerator();
        var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.L);
        var ps = new PostscriptQRCode(data).GetGraphic(new Size(33, 33));
        RemoveCreationDate(ps).ShouldMatchApproved(x => x.NoDiff());
    }

    [Fact]
    public void can_render_postscript_qrcode_size_no_quiet_zones()
    {
        //Create QR code
        var gen = new QRCodeGenerator();
        var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.L);
        var ps = new PostscriptQRCode(data).GetGraphic(new Size(50, 50), false);
        RemoveCreationDate(ps).ShouldMatchApproved(x => x.NoDiff());
    }

    [Fact]
    public void can_render_postscript_qrcode_colors()
    {
        //Create QR code
        var gen = new QRCodeGenerator();
        var data = gen.CreateQrCode("This is a quick test! 123#?", QRCodeGenerator.ECCLevel.L);
        var ps = new PostscriptQRCode(data).GetGraphic(5, Color.Red, Color.Blue);
        RemoveCreationDate(ps).ShouldMatchApproved(x => x.NoDiff());
    }

    private static string RemoveCreationDate(string text)
    {
        // Regex pattern to match lines that start with %%CreationDate: followed by any characters until the end of the line
        string pattern = @"%%CreationDate:.*\r?\n?";

        // Use Regex.Replace to remove matching lines
        return Regex.Replace(text, pattern, string.Empty, RegexOptions.Multiline);
    }
}
