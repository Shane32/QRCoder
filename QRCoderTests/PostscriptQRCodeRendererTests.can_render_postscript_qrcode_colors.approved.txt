%!PS-Adobe-3.0 
%%Creator: QRCoder.NET
%%Title: QRCode
%%DocumentData: Clean7Bit
%%Origin: 0
%%DocumentMedia: Default 165 165 0 () ()
%%BoundingBox: 0 0 165 165
%%LanguageLevel: 2 
%%Pages: 1
%%Page: 1 1
%%EndComments
%%BeginConstants
/sz 165 def
/sc 5 def
%%EndConstants
%%BeginFeature: *PageSize Default
<< /PageSize [ sz sz ] /ImagingBBox null >> setpagedevice
%%EndFeature
%%BeginFunctions 
/csquare {
    newpath
    0 0 moveto
    0 1 rlineto
    1 0 rlineto
    0 -1 rlineto
    closepath
    setrgbcolor
    fill
} def
/f { 
    1 0 0 csquare
    1 0 translate
} def
/b { 
    1 0 translate
} def 
/background { 
    0 0 1 csquare 
} def
/nl {
    -33 -1 translate
} def
%%EndFunctions
%%BeginBody
0 0 moveto
gsave
sz sz scale
background
grestore
gsave
sc sc scale
0 33 1 sub translate
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b f f f f f f f b f b f b f b b b f b f f f f f f f b b b b 
nl
b b b b f b b b b b f b b f f f f b b b b b f b b b b b f b b b b 
nl
b b b b f b f f f b f b f f f b b f f b b b f b f f f b f b b b b 
nl
b b b b f b f f f b f b f f f b f f f f f b f b f f f b f b b b b 
nl
b b b b f b f f f b f b f b b f f b b f f b f b f f f b f b b b b 
nl
b b b b f b b b b b f b b b f f b b b b f b f b b b b b f b b b b 
nl
b b b b f f f f f f f b f b f b f b f b f b f f f f f f f b b b b 
nl
b b b b b b b b b b b b b b f b f b b b b b b b b b b b b b b b b 
nl
b b b b f f f f b b f b f b f f b b f b f f b b f f f b f b b b b 
nl
b b b b f f b f f f b f f b f b f b f b b b b b b f f b b b b b b 
nl
b b b b b f f f f f f b b f f f f b b b f f b f f f b b b b b b b 
nl
b b b b b b b f b f b b f f f f f f b f f b b f b f f b b b b b b 
nl
b b b b f b f f b f f f b f f f b f b b b f f b f b f b b b b b b 
nl
b b b b b f f f b f b b b b b b b b f f f b f f f b b f f b b b b 
nl
b b b b b f b b b f f f f f b b f f b b b b b f b f b f b b b b b 
nl
b b b b f b f f b f b f f f f f f b f b b f f b b f b b f b b b b 
nl
b b b b b b f b f f f f f f f f b f f b f f f f f b f b b b b b b 
nl
b b b b b b b b b b b b f f b f b f b b f b b b f b f f b b b b b 
nl
b b b b f f f f f f f b b b b b b b b b f b f b f f b f f b b b b 
nl
b b b b f b b b b b f b b f f b b b f f f b b b f f b b f b b b b 
nl
b b b b f b f f f b f b b b f b f b b b f f f f f b b b f b b b b 
nl
b b b b f b f f f b f b f b f f f b f b b f f b f f f f f b b b b 
nl
b b b b f b f f f b f b f b b f b b f b f b f b f f b f b b b b b 
nl
b b b b f b b b b b f b f f f f b f f b b f b b f f f b b b b b b 
nl
b b b b f f f f f f f b f b b b b f b b b f f b b b f f f b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
nl
b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b b 
%%EndBody
grestore
showpage   
%%EOF
